lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -227,74 +194,144 @@ Point TruncatedDistribution::getRealization() const
   }
 }
 
 /* Get the DDF of the distribution: DDF_trunc = 1[a, b] * DDF / (CDF(b) - CDF(a)) */
 Point TruncatedDistribution::computeDDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
 
-  const Scalar x = point[0];
-  if ((x <= lowerBound_) || (x > upperBound_)) return Point(1, 0.0);
+  if (!getRange().contains(point)) return Point(getDimension(), 0.0);
   return normalizationFactor_ * distribution_.computeDDF(point);
 }
 
 
 /* Get the PDF of the distribution: PDF_trunc = 1[a, b] * PDF / (CDF(b) - CDF(a)) */
 Scalar TruncatedDistribution::computePDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  if (point.getDimension() != getDimension()) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << getDimension() << ", here dimension=" << point.getDimension();
+
+  if (getDimension() == 1)
+  {
+    const Scalar x = point[0];
+    if ((x < getRange().getLowerBound()[0] - quantileEpsilon_) || (x > getRange().getUpperBound()[0] + quantileEpsilon_)) return 0.0;
+  }
+  else
+    if (!epsilonRange_.contains(point)) return 0.0;
 
-  const Scalar x = point[0];
-  if ((x < lowerBound_ - quantileEpsilon_) || (x > upperBound_ + quantileEpsilon_)) return 0.0;
   return normalizationFactor_ * distribution_.computePDF(point);
 }
 
 
 /* Get the CDF of the distribution: CDF_trunc = 1[a, b] * (CDF - CDF(a)) / (CDF(b) - CDF(a)) + 1]b, inf] */
 Scalar TruncatedDistribution::computeCDF(const Point & point) const
 {
-  if (point.getDimension() != 1) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=1, here dimension=" << point.getDimension();
+  const UnsignedInteger dimension = getDimension();
+  if (point.getDimension() != dimension) throw InvalidArgumentException(HERE) << "Error: the given point must have dimension=" << dimension << ", here dimension=" << point.getDimension();
 
-  const Scalar x = point[0];
-  if (x <= lowerBound_) return 0.0;
-  if (x >= upperBound_) return 1.0;
-  // If tail=true, don't call distribution_.computeCDF with tail=true in the next line!
-  return normalizationFactor_ * (distribution_.computeCDF(point) - cdfLowerBound_);
+  if (getDimension() == 1)
+  {
+    const Scalar x = point[0];
+    if (x <= getRange().getLowerBound()[0]) return 0.0;
+    if (x >= getRange().getUpperBound()[0]) return 1.0;
+
+    // If tail=true, don't call distribution_.computeCDF with tail=true in the next line!
+    return normalizationFactor_ * (distribution_.computeCDF(point) - cdfLowerBound_);
+  }
+
```

Comment:
While being correct, the next lines should be dropped and replaced by
```
return normalizationFactor_ * distribution_.computeProbability(Interval(range_.getLowerBound(), point));
```
as distribution_.computeProbability() should take care of all the corner cases and do the corresponding optimizations.

Reply:
That's dim 1, the multivariate path does contain the right formula.